gneiss/regression/_model.py

```diff
@@ -6,109 +6,40 @@
-
-    @property
-    def r2(self):
-        """ Coefficient of determination for overall fit"""
-        # Reason why we wanted to move this out was because not
-        # all types of statsmodels regressions have this property.
+class RegressionModel(Model):
+    def __init__(self, *args, **kwargs):
+        """
+        Summary object for storing regression results.
 
-        # See `statsmodels.regression.linear_model.RegressionResults`
-        # for more explanation on `ess` and `ssr`.
-        # sum of squares regression. Also referred to as
-        # explained sum of squares.
-        ssr = sum([r.ess for r in self.results])
-        # sum of squares error.  Also referred to as sum of squares residuals
-        sse = sum([r.ssr for r in self.results])
-        # calculate the overall coefficient of determination (i.e. R2)
-        sst = sse + ssr
-        return 1 - sse / sst
+        A `RegressionResults` object stores information about the
+        individual balances used in the regression, the coefficients,
+        residuals. This object can be used to perform predictions.
+        In addition, summary statistics such as the coefficient
+        of determination for the overall fit can be calculated.
 
-    def _check_projection(self, project):
-        """ Checks to make sure that the `ilr_inv` can be performed.
 
         Parameters
         ----------
-        project : bool
-           Specifies if a projection into the Aitchison simplex can be
-           performed.
-
-        Raises
-        ------
-        ValueError:
-            Cannot perform projection into Aitchison simplex if `basis`
-            is not specified.
-        ValueError:
-            Cannot perform projection into Aitchison simplex
-            if `feature_names` is not specified.
+        submodels : list of statsmodels objects
+            List of statsmodels result objects.
+        basis : pd.DataFrame
+            Orthonormal basis in the Aitchison simplex.
+            Row names correspond to the leafs of the tree
+            and the column names correspond to the internal nodes
+            in the tree. If this is not specified, then `project` cannot
+            be enabled in `coefficients` or `predict`.
+        tree : skbio.TreeNode
+            Bifurcating tree that defines `basis`.
+        balances : pd.DataFrame
+            A table of balances where samples are rows and
+            balances are columns.  These balances were calculated
+            using `tree`.
         """
-        if self.basis is None and project:
-            raise ValueError("Cannot perform projection into Aitchison simplex"
-                             " if `basis` is not specified.")
-
-        if self.feature_names is None and project:
-            raise ValueError("Cannot perform projection into Aitchison simplex"
-                             " if `feature_names` is not specified.")
+        super().__init__(*args, **kwargs)
```

Comment:
Exactly!  In future iterations of gneiss, we'll want to have support for other types of models such as Classifiers (SVM, Logistic, ...) and Cross correlation (Partial Least Squares, Canonical Correlation Analysis, ...) 

Having this placeholder is critical since there are specific abstract methods that will be common across all model objects (i.e. serialization)